### PR TITLE
confgen: export deterministic JSON/Text/Bin

### LINKED ISSUE #61: confgen: export deterministic JSON/Text/Bin
## Problems

Protobuf `map` type is a hashmap, so elements are not ordered and non-deterministic.

- [encoding/protojson#Marshal](https://pkg.go.dev/google.golang.org/protobuf/encoding/protojson#Marshal)
Marshal writes the given proto.Message in JSON format using default options. **Do not depend on the output being stable**. It may change over time across different versions of the program.
- [encoding/prototext#Marshal](https://pkg.go.dev/google.golang.org/protobuf/encoding/prototext#Marshal)
Marshal writes the given proto.Message in textproto format using default options. **Do not depend on the output being stable**. It may change over time across different versions of the program.
- [encoding/protowire#Marshal](https://pkg.go.dev/google.golang.org/protobuf/proto#Marshal)

### Related issues
- https://github.com/golang/protobuf/issues/1373
- https://github.com/bufbuild/connect-go/pull/460
- https://protobuf.dev/reference/go/faq/#unstable-json
- https://protobuf.dev/reference/go/faq/#unstable-text

## Solutions
- [x] JSON: to gain some degree of output stability, we recommend running the output through a JSON formatter.
- [x] Text: To obtain some degree of stability, we recommend passing the output of prototext through the [txtpbfmt](https://github.com/protocolbuffers/txtpbfmt) program. The formatter can be directly invoked in Go using [parser.Format](https://pkg.go.dev/github.com/protocolbuffers/txtpbfmt/parser?tab=doc#Format).
- [x] Bin: set marshal option `Deterministic` as `true`. See [MarshalOptions](https://github.com/protocolbuffers/protobuf-go/blob/v1.30.0/proto/encode.go#L21).

### PR DESCRIPTION
**Pull request by wenchy**
- close #61

### CODE REVIEW

File: internal/confgen/mexporter/marshal.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		return nil, err
+	}
+	if options.Pretty {
+		prettyJSON := new(bytes.Buffer)
+		if err := json.Indent(prettyJSON, stableJSON.Bytes(), "", "    "); err != nil {
+			return nil, err
+		}
+		return prettyJSON.Bytes(), nil
+	}
+	return stableJSON.Bytes(), nil
```
Comment (huieric) [inline internal/confgen/mexporter/marshal.go:77]: Add tests to cover cases not using `Pretty` option.

File: internal/confgen/mexporter/mexporter.go [hunk 2]
```diff
@@ -59,19 +57,25 @@ func (x *messageExporter) export(fmt format.Format) error {
 	switch fmt {
 	case format.JSON:
 		filename += format.JSONExt
-		out, err = x.marshalToJSON()
+		options := &MarshalOptions{
+			Pretty:          x.outputOpt.Pretty,
+			EmitUnpopulated: x.outputOpt.EmitUnpopulated,
+			UseProtoNames:   x.outputOpt.UseProtoNames,
+			UseEnumNumbers:  x.outputOpt.UseEnumNumbers,
+		}
```
Comment (huieric) [inline internal/confgen/mexporter/mexporter.go:65]: Add tests to cover these new lines.

File: internal/confgen/mexporter/marshal.go (hunk 1 above)
Comment (wenchy) [reply internal/confgen/mexporter/marshal.go:78]: Done

File: internal/confgen/mexporter/mexporter.go (hunk 2 above)
Comment (wenchy) [reply internal/confgen/mexporter/mexporter.go:65]: Done

**Pull request merged**